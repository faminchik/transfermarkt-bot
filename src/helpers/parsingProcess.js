import _ from 'lodash';
import convertData from 'helpers/convertData';
import getTableDataFromHTML from 'helpers/getTableDataFromHTML';
import ConvertDataConfig from 'configs/ConvertDataConfig';
import ConvertDataStrategies from 'configs/ConvertDataStrategies';

export default (html, types = []) =>
    _.reduce(
        types,
        (result, type) => {
            const { textData, htmlData } = getTableDataFromHTML(html, type);
            if (!textData && !htmlData) {
                result[type] = [];
                return result;
            }

            const strategy = ConvertDataStrategies[type];
            const config = ConvertDataConfig[strategy];
            const convertedData = convertData({ textData, htmlData }, config);

            result[type] = convertedData;
            return result;
        },
        {}
    );
