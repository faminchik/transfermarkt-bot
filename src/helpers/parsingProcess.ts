import convertData from 'helpers/convertData';
import getTableDataFromHTML from 'helpers/getTableDataFromHTML';
import type { TConvertedDataMapper } from 'ts/ConvertDataTS';
import type pt from 'constants/transfermarkt/ParsingTypes';

export default <T extends pt>(html: string, type: T): TConvertedDataMapper[T] => {
    const { textData, htmlData } = getTableDataFromHTML(html, type);

    return convertData({ textData, htmlData }, type);
};
