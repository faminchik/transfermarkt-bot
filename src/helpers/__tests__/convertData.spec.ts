import parsingProcess from 'helpers/parsingProcess';
import pt from 'constants/transfermarkt/ParsingTypes';

import allLatestTransfersHtml from './data/allLatestTransfers.data';
import searchResultHtml from './data/searchResult.data';
import teamTransfersHtml from './data/teamTransfers.data';

describe('convertData', () => {
    describe('allLatestTransfers', () => {
        test('should convert html data in a correct way', () => {
            const transfersInfo = parsingProcess(allLatestTransfersHtml, pt.ALL_LATEST_TRANSFERS);

            expect(transfersInfo).toMatchSnapshot();
        });
    });

    describe('search', () => {
        describe('clubsSearch', () => {
            test('should convert html data in a correct way', () => {
                const clubs = parsingProcess(searchResultHtml, pt.SEARCH_CLUBS);

                expect(clubs).toMatchSnapshot();
            });
        });

        describe('playersSearch', () => {
            test('should convert html data in a correct way', () => {
                const clubs = parsingProcess(searchResultHtml, pt.SEARCH_PLAYERS);

                expect(clubs).toMatchSnapshot();
            });
        });
    });

    describe('teamTransfers', () => {
        describe('arrivals transfers', () => {
            test('should convert html data in a correct way', () => {
                const arrivalsTeamTransfers = parsingProcess(teamTransfersHtml, pt.TEAM_TRANSFERS_ARRIVALS);

                expect(arrivalsTeamTransfers).toMatchSnapshot();
            });
        });

        describe('departures transfers', () => {
            test('should convert html data in a correct way', () => {
                const departuresTeamTransfers = parsingProcess(teamTransfersHtml, pt.TEAM_TRANSFERS_DEPARTURES);

                expect(departuresTeamTransfers).toMatchSnapshot();
            });
        });
    });
});
