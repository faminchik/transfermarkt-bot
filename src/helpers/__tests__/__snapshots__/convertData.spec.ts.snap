// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertData allLatestTransfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "32",
    "fee": "?",
    "joinedTeam": "Toronto FC",
    "joinedTeamCountry": "Canada",
    "leftTeam": "AJ Auxerre",
    "leftTeamCountry": "France",
    "marketValue": "€600k",
    "name": "Quentin Westberg",
    "nationality": "United States",
    "profileLink": "/quentin-westberg/profil/spieler/33934",
    "transferDate": "Feb 25, 2019",
  },
  Object {
    "age": "27",
    "fee": "Loan",
    "joinedTeam": "CFR Cluj",
    "joinedTeamCountry": "Romania",
    "leftTeam": "GZ Hengfeng",
    "leftTeamCountry": "China",
    "marketValue": "€1.20m",
    "name": "Kévin Boli",
    "nationality": "Cote d'Ivoire",
    "profileLink": "/kevin-boli/profil/spieler/186593",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "22",
    "fee": "?",
    "joinedTeam": "BJ Sinobo Guoan",
    "joinedTeamCountry": "China",
    "leftTeam": "West Brom",
    "leftTeamCountry": "England",
    "marketValue": "€1.50m",
    "name": "Yuning Zhang",
    "nationality": "China",
    "profileLink": "/yuning-zhang/profil/spieler/264429",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "27",
    "fee": "€450k",
    "joinedTeam": "GZ Hengfeng",
    "joinedTeamCountry": "China",
    "leftTeam": "Apol. Limassol",
    "leftTeamCountry": "Cyprus",
    "marketValue": "€1.50m",
    "name": "Anton Maglica",
    "nationality": "Croatia",
    "profileLink": "/anton-maglica/profil/spieler/113192",
    "transferDate": "Feb 24, 2019",
  },
  Object {
    "age": "30",
    "fee": "?",
    "joinedTeam": "BJ Renhe",
    "joinedTeamCountry": "China",
    "leftTeam": "Reading",
    "leftTeamCountry": "England",
    "marketValue": "€3.50m",
    "name": "Sone Aluko",
    "nationality": "Nigeria",
    "profileLink": "/sone-aluko/profil/spieler/42198",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "30",
    "fee": "",
    "joinedTeam": "Chicago Fire",
    "joinedTeamCountry": "United States",
    "leftTeam": "Philadelphia",
    "leftTeamCountry": "United States",
    "marketValue": "€1.25m",
    "name": "C.J. Sapong",
    "nationality": "United States",
    "profileLink": "/c-j-sapong/profil/spieler/174724",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "28",
    "fee": "-",
    "joinedTeam": "Újpest FC",
    "joinedTeamCountry": "Hungary",
    "leftTeam": "Without Club",
    "leftTeamCountry": "",
    "marketValue": "€2.50m",
    "name": "Lacina Traoré",
    "nationality": "Cote d'Ivoire",
    "profileLink": "/lacina-traore/profil/spieler/69006",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "31",
    "fee": "Free Transfer",
    "joinedTeam": "Atyrau",
    "joinedTeamCountry": "Kazakhstan",
    "leftTeam": "Akzhayik",
    "leftTeamCountry": "Kazakhstan",
    "marketValue": "€600k",
    "name": "Andrii Tkachuk",
    "nationality": "Ukraine",
    "profileLink": "/andrii-tkachuk/profil/spieler/58150",
    "transferDate": "Feb 23, 2019",
  },
  Object {
    "age": "28",
    "fee": "€2.50m",
    "joinedTeam": "Grêmio",
    "joinedTeamCountry": "Brazil",
    "leftTeam": "Sport Recife",
    "leftTeamCountry": "Brazil",
    "marketValue": "€2.75m",
    "name": "André",
    "nationality": "Brazil",
    "profileLink": "/andre/profil/spieler/104509",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "27",
    "fee": "-",
    "joinedTeam": "Without Club",
    "joinedTeamCountry": "",
    "leftTeam": "KS Samara",
    "leftTeamCountry": "Russia",
    "marketValue": "€700k",
    "name": "Alan Chochiev",
    "nationality": "Russia",
    "profileLink": "/alan-chochiev/profil/spieler/104511",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "Loan",
    "joinedTeam": "Arsenal Kyiv",
    "joinedTeamCountry": "Ukraine",
    "leftTeam": "Shakhtar D.",
    "leftTeamCountry": "Ukraine",
    "marketValue": "€500k",
    "name": "Viacheslav Tankovskyi",
    "nationality": "Ukraine",
    "profileLink": "/viacheslav-tankovskyi/profil/spieler/213980",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "22",
    "fee": "?",
    "joinedTeam": "Pakhtakor",
    "joinedTeamCountry": "Uzbekistan",
    "leftTeam": "Kokand 1912",
    "leftTeamCountry": "Uzbekistan",
    "marketValue": "€600k",
    "name": "Javokhir Sidikov",
    "nationality": "Uzbekistan",
    "profileLink": "/javokhir-sidikov/profil/spieler/288957",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "24",
    "fee": "Loan",
    "joinedTeam": "AFC Eskilstuna",
    "joinedTeamCountry": "Sweden",
    "leftTeam": "Rayo Vallecano",
    "leftTeamCountry": "Spain",
    "marketValue": "€800k",
    "name": "José León",
    "nationality": "Spain",
    "profileLink": "/jose-leon/profil/spieler/281773",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "28",
    "fee": "?",
    "joinedTeam": "Mladá Boleslav",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "€600k",
    "name": "Muris Mesanovic",
    "nationality": "Bosnia-Herzegovina",
    "profileLink": "/muris-mesanovic/profil/spieler/70176",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "25",
    "fee": "Loan",
    "joinedTeam": "Slovan Liberec",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "€1.25m",
    "name": "Jan Sykora",
    "nationality": "Czech Republic",
    "profileLink": "/jan-sykora/profil/spieler/187490",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "26",
    "fee": "Loan",
    "joinedTeam": "Mladá Boleslav",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "Slavia Prag",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "€1.25m",
    "name": "Jakub Jugas",
    "nationality": "Czech Republic",
    "profileLink": "/jakub-jugas/profil/spieler/146712",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "?",
    "joinedTeam": "Sparta Praha",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "MFK Karvina",
    "leftTeamCountry": "Czech Republic",
    "marketValue": "€800k",
    "name": "Filip Panak",
    "nationality": "Czech Republic",
    "profileLink": "/filip-panak/profil/spieler/331963",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "26",
    "fee": "Loan",
    "joinedTeam": "Sochi",
    "joinedTeamCountry": "Russia",
    "leftTeam": "Loko Moscow",
    "leftTeamCountry": "Russia",
    "marketValue": "€500k",
    "name": "Timofey Margasov",
    "nationality": "Russia",
    "profileLink": "/timofey-margasov/profil/spieler/148485",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "23",
    "fee": "?",
    "joinedTeam": "FC Astana",
    "joinedTeamCountry": "Kazakhstan",
    "leftTeam": "Club Brugge",
    "leftTeamCountry": "Belgium",
    "marketValue": "€1.00m",
    "name": "Dorin Rotariu",
    "nationality": "Romania",
    "profileLink": "/dorin-rotariu/profil/spieler/250573",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "33",
    "fee": "End of loan",
    "joinedTeam": "Birmingham",
    "joinedTeamCountry": "England",
    "leftTeam": "Coventry",
    "leftTeamCountry": "England",
    "marketValue": "€1.00m",
    "name": "David Stockdale",
    "nationality": "England",
    "profileLink": "/david-stockdale/profil/spieler/67339",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "30",
    "fee": "€1.30m",
    "joinedTeam": "Sparta Praha",
    "joinedTeamCountry": "Czech Republic",
    "leftTeam": "HN Jianye",
    "leftTeamCountry": "China",
    "marketValue": "€2.00m",
    "name": "Borek Dockal",
    "nationality": "Czech Republic",
    "profileLink": "/borek-dockal/profil/spieler/62983",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "29",
    "fee": "Free Transfer",
    "joinedTeam": "KS Samara",
    "joinedTeamCountry": "Russia",
    "leftTeam": "Anzhi",
    "leftTeamCountry": "Russia",
    "marketValue": "€600k",
    "name": "Mohammed Rabiu",
    "nationality": "Ghana",
    "profileLink": "/mohammed-rabiu/profil/spieler/67721",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "22",
    "fee": "End of loan",
    "joinedTeam": "West Brom",
    "joinedTeamCountry": "England",
    "leftTeam": "ADO Den Haag",
    "leftTeamCountry": "Netherlands",
    "marketValue": "€1.50m",
    "name": "Yuning Zhang",
    "nationality": "China",
    "profileLink": "/yuning-zhang/profil/spieler/264429",
    "transferDate": "Feb 22, 2019",
  },
  Object {
    "age": "24",
    "fee": "Loan",
    "joinedTeam": "Poli. Iasi",
    "joinedTeamCountry": "Romania",
    "leftTeam": "Levski Sofia",
    "leftTeamCountry": "Bulgaria",
    "marketValue": "€700k",
    "name": "Filipe Nascimento",
    "nationality": "Portugal",
    "profileLink": "/filipe-nascimento/profil/spieler/197999",
    "transferDate": "Feb 21, 2019",
  },
  Object {
    "age": "25",
    "fee": "€90k",
    "joinedTeam": "Sepsi Sf. Gh.",
    "joinedTeamCountry": "Romania",
    "leftTeam": "FC Poli Iasi",
    "leftTeamCountry": "Romania",
    "marketValue": "€700k",
    "name": "Dylan Flores",
    "nationality": "Costa Rica",
    "profileLink": "/dylan-flores/profil/spieler/131250",
    "transferDate": "Feb 21, 2019",
  },
]
`;

exports[`convertData search clubsSearch should convert html data in a correct way 1`] = `
Array [
  Object {
    "clubLink": "/inter-mailand/startseite/verein/46",
    "clubName": "Inter Milan",
    "country": "Italy",
    "totalMarketValue": "€557.40m",
  },
  Object {
    "clubLink": "/ac-mailand/startseite/verein/5",
    "clubName": "AC Milan",
    "country": "Italy",
    "totalMarketValue": "€530.00m",
  },
  Object {
    "clubLink": "/fk-metalac-gornji-milanovac/startseite/verein/12135",
    "clubName": "FK Metalac Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "€3.05m",
  },
  Object {
    "clubLink": "/inter-mailand-primavera/startseite/verein/5380",
    "clubName": "Inter Milan Primavera",
    "country": "Italy",
    "totalMarketValue": "€2.20m",
  },
  Object {
    "clubLink": "/inter-mailand-uefa-u19/startseite/verein/62104",
    "clubName": "Inter Mailand UEFA U19",
    "country": "Italy",
    "totalMarketValue": "€2.20m",
  },
  Object {
    "clubLink": "/us-bustese-roncalli/startseite/verein/49933",
    "clubName": "Milano City FC",
    "country": "Italy",
    "totalMarketValue": "€700k",
  },
  Object {
    "clubLink": "/ac-mailand-primavera/startseite/verein/10958",
    "clubName": "Milan Primavera",
    "country": "Italy",
    "totalMarketValue": "€400k",
  },
  Object {
    "clubLink": "/milano-united-a-f-c-/startseite/verein/37588",
    "clubName": "Milano Unite FC",
    "country": "South Africa",
    "totalMarketValue": "€250k",
  },
  Object {
    "clubLink": "/fk-takovo-gornji-milanovac/startseite/verein/32117",
    "clubName": "FK Takovo Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "€100k",
  },
  Object {
    "clubLink": "/fk-lunjevica-gornji-milanovac/startseite/verein/71902",
    "clubName": "FK Lunjevica Gornji Milanovac",
    "country": "Serbia",
    "totalMarketValue": "€100k",
  },
]
`;

exports[`convertData teamTransfers arrivals transfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "19",
    "fee": "€135.00m",
    "marketValue": "€120.00m",
    "name": "Kylian Mbappé",
    "nationality": "France",
    "secondPartyTeam": "Monaco",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "21",
    "fee": "€37.00m",
    "marketValue": "€18.00m",
    "name": "Thilo Kehrer",
    "nationality": "Germany",
    "secondPartyTeam": "FC Schalke 04",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "25",
    "fee": "€5.00m",
    "marketValue": "€12.00m",
    "name": "Juan Bernat",
    "nationality": "Spain",
    "secondPartyTeam": "Bayern Munich",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "40",
    "fee": "Free Transfer",
    "marketValue": "€2.00m",
    "name": "Gianluigi Buffon",
    "nationality": "Italy",
    "secondPartyTeam": "Juventus",
    "secondPartyTeamCountry": "Italy",
  },
  Object {
    "age": "29",
    "fee": "Free Transfer",
    "marketValue": "€8.00m",
    "name": "Eric Maxim Choupo-Moting",
    "nationality": "Cameroon",
    "secondPartyTeam": "Stoke City",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Colin Dagba",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Antoine Bernede",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "19",
    "fee": "-",
    "marketValue": "-",
    "name": "Stanley N'Soki",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "18",
    "fee": "-",
    "marketValue": "€500k",
    "name": "Moussa Diaby",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "18",
    "fee": "-",
    "marketValue": "€1.00m",
    "name": "Timothy Weah",
    "nationality": "United States",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "17",
    "fee": "-",
    "marketValue": "-",
    "name": "Yacine Adli",
    "nationality": "France",
    "secondPartyTeam": "Paris SG B",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "20",
    "fee": "End of loan",
    "marketValue": "€700k",
    "name": "Jonathan Ikoné",
    "nationality": "France",
    "secondPartyTeam": "Montpellier",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "28",
    "fee": "End of loan",
    "marketValue": "€18.00m",
    "name": "Grzegorz Krychowiak",
    "nationality": "Poland",
    "secondPartyTeam": "West Brom",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "22",
    "fee": "End of loan",
    "marketValue": "€600k",
    "name": "Rémy Descamps",
    "nationality": "France",
    "secondPartyTeam": "Tours FC",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "25",
    "fee": "End of loan",
    "marketValue": "€7.50m",
    "name": "Jesé",
    "nationality": "Spain",
    "secondPartyTeam": "Stoke City",
    "secondPartyTeamCountry": "England",
  },
  Object {
    "age": "21",
    "fee": "End of loan",
    "marketValue": "€40.00m",
    "name": "Gonçalo Guedes",
    "nationality": "Portugal",
    "secondPartyTeam": "Valencia",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "25",
    "fee": "End of loan",
    "marketValue": "€1.50m",
    "name": "Jean-Christophe Bahebeck",
    "nationality": "France",
    "secondPartyTeam": "FC Utrecht",
    "secondPartyTeamCountry": "Netherlands",
  },
  Object {
    "age": "20",
    "fee": "End of loan",
    "marketValue": "€1.50m",
    "name": "Odsonne Edouard",
    "nationality": "France",
    "secondPartyTeam": "Celtic",
    "secondPartyTeamCountry": "Scotland",
  },
]
`;

exports[`convertData teamTransfers departures transfers should convert html data in a correct way 1`] = `
Array [
  Object {
    "age": "21",
    "fee": "€40.00m",
    "marketValue": "€40.00m",
    "name": "Gonçalo Guedes",
    "nationality": "Portugal",
    "secondPartyTeam": "Valencia",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "29",
    "fee": "€24.70m",
    "marketValue": "€15.00m",
    "name": "Javier Pastore",
    "nationality": "Argentina",
    "secondPartyTeam": "AS Roma",
    "secondPartyTeamCountry": "Italy",
  },
  Object {
    "age": "28",
    "fee": "€24.00m",
    "marketValue": "€17.00m",
    "name": "Yuri Berchiche",
    "nationality": "Spain",
    "secondPartyTeam": "Athletic",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "20",
    "fee": "€10.30m",
    "marketValue": "€1.50m",
    "name": "Odsonne Edouard",
    "nationality": "France",
    "secondPartyTeam": "Celtic",
    "secondPartyTeamCountry": "Scotland",
  },
  Object {
    "age": "20",
    "fee": "€5.00m",
    "marketValue": "€700k",
    "name": "Jonathan Ikoné",
    "nationality": "France",
    "secondPartyTeam": "LOSC Lille",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "22",
    "fee": "Loan fee:€2.00m",
    "marketValue": "€30.00m",
    "name": "Giovani Lo Celso",
    "nationality": "Argentina",
    "secondPartyTeam": "Real Betis",
    "secondPartyTeamCountry": "Spain",
  },
  Object {
    "age": "28",
    "fee": "Loan fee:€1.00m",
    "marketValue": "€18.00m",
    "name": "Grzegorz Krychowiak",
    "nationality": "Poland",
    "secondPartyTeam": "Loko Moscow",
    "secondPartyTeamCountry": "Russia",
  },
  Object {
    "age": "25",
    "fee": "Free Transfer",
    "marketValue": "€1.50m",
    "name": "Jean-Christophe Bahebeck",
    "nationality": "France",
    "secondPartyTeam": "FC Utrecht",
    "secondPartyTeamCountry": "Netherlands",
  },
  Object {
    "age": "28",
    "fee": "Loan",
    "marketValue": "€8.00m",
    "name": "Kevin Trapp",
    "nationality": "Germany",
    "secondPartyTeam": "E. Frankfurt",
    "secondPartyTeamCountry": "Germany",
  },
  Object {
    "age": "22",
    "fee": "Loan",
    "marketValue": "€600k",
    "name": "Rémy Descamps",
    "nationality": "France",
    "secondPartyTeam": "Clermont Foot",
    "secondPartyTeamCountry": "France",
  },
  Object {
    "age": "31",
    "fee": "-",
    "marketValue": "€5.00m",
    "name": "Hatem Ben Arfa",
    "nationality": "France",
    "secondPartyTeam": "Without Club",
    "secondPartyTeamCountry": "",
  },
  Object {
    "age": "35",
    "fee": "-",
    "marketValue": "€1.00m",
    "name": "Thiago Motta",
    "nationality": "Italy",
    "secondPartyTeam": "Retired",
    "secondPartyTeamCountry": "",
  },
  Object {
    "age": "19",
    "fee": "End of loan",
    "marketValue": "€120.00m",
    "name": "Kylian Mbappé",
    "nationality": "France",
    "secondPartyTeam": "Monaco",
    "secondPartyTeamCountry": "France",
  },
]
`;
